scripts/coccinelle/bool_function_type.cocci

```diff
@@ -0,0 +1,19 @@
+@@
+identifier fn;
+typedef bool;
+symbol false;
+symbol true;
+@@
+
+- int
```

Comment:
Can you explain this patch for my own education? As I read it, this changes the return type of functions that return boolean literals from int to bool, but that isn't what I see in the diff.

Reply:
Nevermind, I didn't see the other one.

Reply:
This just looks for `int` type functions and checks if it returns true or false. If returns true/false then convert this function to bool type.